src/sage/combinat/permutation.py

```diff
@@ -486,11 +529,14 @@ def __classcall_private__(cls, l, check=True):
             else:
                 raise ValueError("cannot convert l (= %s) to a Permutation" % l)
 
+        if algorithm != "lex" and algorithm != "sjt":
+            raise ValueError("Unsupported algorithm %s; expected 'lex' or 'sjt'"
+            % algorithm)
```

Comment:
This duplicates the check below so remove this

Reply:
This checks for legal a `algorithm` value. The check below (redundant I agree, is done to init or not with an `SJT` object and because the input is checked before, the two conditions are in different parts of the code.